robot/rospackages/src/mcu_control/scripts/CommsNode.py

```diff
@@ -117,40 +113,44 @@ def send_command(command_name, args, deviceToSendTo):
             elif data_type == dt.ARG_FLOAT32_ID:
                 ser.write(float(data)) #perhaps will fuck up
 
-        # make sure to also send the number of bytes
-
         # if args is not None and number_of_arguments != 0:
         #     arg_bytes = get_arg_bytes(args)
         #     for arg_byte in arg_bytes:
         #         ser.write(arg_byte)
         #     ser.write(STOP_BYTE)
         # return True
+
+        # might need to flush serial if buffer gets overloaded
     return False
 
+
 def arm_command_callback(message):
     rospy.loginfo('received: ' + message.data + ' command, sending to arm Teensy')
     command, args = parse_command(message)
 
-    temp_struct = [command, args, ARM_SELECTED]
-    arm_queue.append(temp_struct)
-
-def parse_command(message):
-    full_command = message.split(" ")
-    if full_command is not None:
-        command = full_command[0]
-        args = full_command[1:]
-        newArgs = []
-        for arg in args:
-            try:
-                newArgs.append(float(arg))
-            except ValueError:
-                newArgs.append(arg)
-
-        return command, newArgs
-    return None, []
-
-def get_arg_bytes(command_tuple):
-    return sum(element[1] for element in command_tuple[2])
+    temp_list = [command, args, ARM_SELECTED]
+    arm_queue.append(temp_list)
+
+
+def main():
+    try:
+        while not rospy.is_shutdown():
+            # Sends a command to each teensy if there is anything to send 
+            for queue in command_queues:
+                if len(queue) > 0:
+                    cmd = queue.pop()
+                    send_command(cmd)
+
+            # Prompts each teensy to send something
+            for teensy in teensy_pins:
+                GPIO.output(teensy, 1)
+                receive_message()
+                GPIO.output(teensy, 0)
```

Comment:
receive_message calls a while(True): loop, are you sure you want to do that lol?

Reply:
No, not sure. I'll set a timeout.

Reply:
ok made it wait for half a second, hopefully that is plenty. I'm not sure if there is a better way. 
Also, maybe we can just set a global timeout when starting serial?